python/pyarrow/dataset.py

```diff
@@ -622,62 +631,74 @@ def dataset(source, schema=None, format=None, filesystem=None,
 
     Examples
     --------
+    Creating an example pa.Table:
+
+    >>> import pyarrow as pa
+    >>> import pyarrow.parquet as pq
+    >>> table = pa.table({'year': [2020, 2022, 2021, 2022, 2019, 2021],
+    ...                   'n_legs': [2, 2, 4, 4, 5, 100],
+    ...                   'animal': ["Flamingo", "Parrot", "Dog", "Horse",
+    ...                              "Brittle stars", "Centipede"]})
+    >>> pq.write_table(table, "file.parquet")
```

Comment:
Wondering: those files, do they end up in the directory from where you are running pytest / where the dataset.py is located? 

Because if so, we will probably want to clean them up in some way, so you don't get a bunch of files in your repo from running the doctests. 
(it might be possible to change python's "current working directory" to a temporary directory in conftest.py?)